From cfc7f0a0070a3919aa4a0293278369e6db717aab Mon Sep 17 00:00:00 2001
From: Thomas Zimmermann <tzimmermann@suse.de>
Date: Wed, 18 May 2022 10:44:06 +0200
Subject: [PATCH 092/105] xf86: Accept devices with the kernel's ofdrm driver

Add a workaround to accept devices of the kernel's ofdrm driver.
Makes Xorg work on Open Firmware's pre-configured display with the
DRM graphics stack.

Signed-off-by: Thomas Zimmermann <tzimmermann@suse.de>
---
 hw/xfree86/common/xf86platformBus.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/hw/xfree86/common/xf86platformBus.c b/hw/xfree86/common/xf86platformBus.c
index 1d1f87c74..4cb9a671d 100644
--- a/hw/xfree86/common/xf86platformBus.c
+++ b/hw/xfree86/common/xf86platformBus.c
@@ -579,6 +579,9 @@ xf86platformProbeDev(DriverPtr drvp)
                     /* Accept the device if the driver is hyperv_drm */
                     if (strcmp(xf86_platform_devices[j].attribs->driver, "hyperv_drm") == 0)
                         break;
+                    /* Accept the device if the driver is ofdrm */
+                    if (strcmp(xf86_platform_devices[j].attribs->driver, "ofdrm") == 0)
+                        break;
                     /* Accept the device if the driver is simpledrm */
                     if (strcmp(xf86_platform_devices[j].attribs->driver, "simpledrm") == 0)
                         break;
-- 
2.45.2

